test/core/transport/chttp2/frame_test.cc

```diff
@@ -0,0 +1,393 @@
+                                     SliceBufferFromString("kids")}),
+            ByteVec(0, 0, 4, 0, 1, 0x98, 0x38, 0x18, 0x22, 'k', 'i', 'd', 's'));
+  EXPECT_EQ(Serialize(Http2HeaderFrame{1, false, false,
+                                       SliceBufferFromString("hello")}),
+            ByteVec(0, 0, 5, 1, 0, 0, 0, 0, 1, 'h', 'e', 'l', 'l', 'o'));
+  EXPECT_EQ(Serialize(Http2HeaderFrame{1, true, false,
+                                       SliceBufferFromString("hello")}),
+            ByteVec(0, 0, 5, 1, 4, 0, 0, 0, 1, 'h', 'e', 'l', 'l', 'o'));
+  EXPECT_EQ(Serialize(Http2HeaderFrame{1, false, true,
+                                       SliceBufferFromString("hello")}),
+            ByteVec(0, 0, 5, 1, 1, 0, 0, 0, 1, 'h', 'e', 'l', 'l', 'o'));
+  EXPECT_EQ(Serialize(Http2HeaderFrame{1, true, true,
+                                       SliceBufferFromString("hello")}),
+            ByteVec(0, 0, 5, 1, 5, 0, 0, 0, 1, 'h', 'e', 'l', 'l', 'o'));
+  EXPECT_EQ(Serialize(Http2ContinuationFrame{1, false,
+                                             SliceBufferFromString("hello")}),
+            ByteVec(0, 0, 5, 9, 0, 0, 0, 0, 1, 'h', 'e', 'l', 'l', 'o'));
+  EXPECT_EQ(Serialize(Http2ContinuationFrame{1, true,
+                                             SliceBufferFromString("hello")}),
+            ByteVec(0, 0, 5, 9, 4, 0, 0, 0, 1, 'h', 'e', 'l', 'l', 'o'));
+  EXPECT_EQ(Serialize(Http2RstStreamFrame{1, GRPC_HTTP2_CONNECT_ERROR}),
+            ByteVec(0, 0, 4, 3, 0, 0, 0, 0, 1, 0, 0, 0, 0x0a));
+  EXPECT_EQ(Serialize(Http2SettingsFrame{}),
+            ByteVec(0, 0, 0, 4, 0, 0, 0, 0, 0));
+  EXPECT_EQ(
+      Serialize(Http2SettingsFrame{false, {{0x1234, 0x9abcdef0}}}),
+      ByteVec(0, 0, 6, 4, 0, 0, 0, 0, 0, 0x12, 0x34, 0x9a, 0xbc, 0xde, 0xf0));
+  EXPECT_EQ(Serialize(Http2SettingsFrame{
+                false, {{0x1234, 0x9abcdef0}, {0x4321, 0x12345678}}}),
+            ByteVec(0, 0, 12, 4, 0, 0, 0, 0, 0, 0x12, 0x34, 0x9a, 0xbc, 0xde,
+                    0xf0, 0x43, 0x21, 0x12, 0x34, 0x56, 0x78));
+  EXPECT_EQ(Serialize(Http2SettingsFrame{true, {}}),
+            ByteVec(0, 0, 0, 4, 1, 0, 0, 0, 0));
+  EXPECT_EQ(Serialize(Http2PingFrame{false, 0x123456789abcdef0}),
+            ByteVec(0, 0, 8, 6, 0, 0, 0, 0, 0, 0x12, 0x34, 0x56, 0x78, 0x9a,
+                    0xbc, 0xde, 0xf0));
+  EXPECT_EQ(Serialize(Http2PingFrame{true, 0x123456789abcdef0}),
+            ByteVec(0, 0, 8, 6, 1, 0, 0, 0, 0, 0x12, 0x34, 0x56, 0x78, 0x9a,
+                    0xbc, 0xde, 0xf0));
+  EXPECT_EQ(Serialize(Http2GoawayFrame{0x12345678, GRPC_HTTP2_ENHANCE_YOUR_CALM,
+                                       Slice::FromCopiedString("hello")}),
+            ByteVec(0, 0, 13, 7, 0, 0, 0, 0, 0, 0x12, 0x34, 0x56, 0x78, 0, 0, 0,
+                    0x0b, 'h', 'e', 'l', 'l', 'o'));
+  EXPECT_EQ(Serialize(Http2WindowUpdateFrame{1, 0x12345678}),
+            ByteVec(0, 0, 4, 8, 0, 0, 0, 0, 1, 0x12, 0x34, 0x56, 0x78));
+}
+
+TEST(Frame, Parse) {
```

Comment:
can we add some parsing for unknown type frames, push promise and priority type frames as well? (to make sure that they are ignored)